src/test/scala/is/hail/expr/ir/PruneSuite.scala

```diff
@@ -0,0 +1,354 @@
+package is.hail.expr.ir
+
+import is.hail.SparkSuite
+import is.hail.expr._
+import is.hail.expr.types._
+import is.hail.table.Table
+import is.hail.utils._
+import is.hail.variant.MatrixTable
+import org.apache.spark.sql.Row
+import org.testng.annotations.Test
+
+class PruneSuite extends SparkSuite {
+  @Test def testUnionType() {
+    val base = TStruct(
+      "a" -> TStruct(
+        "aa" -> TInt32(),
+        "ab" -> TStruct(
+          "aaa" -> TString()
+        )
+      ),
+      "b" -> TInt32(),
+      "c" -> TArray(TStruct(
+        "ca" -> TInt32()
+      ))
+    )
+
+    assert(PruneDeadFields.unify(base, TStruct()) == TStruct())
+    assert(PruneDeadFields.unify(base, TStruct("b" -> TInt32())) == TStruct("b" -> TInt32()))
```

Comment:
While we're on the subject of concise code, what do you think about a Scala string interpolator for types?

```
  assert(PruneDeadFields.unify(base, t"struct{}") == t"struct{}")
  assert(PruneDeadFields.unify(base, t"struct{b: int32}") == "struct{b: int32}")
  ...
```

Reply:
it does make the Python a lot nicer.

Reply:
we can probably even find a scala parsing library that uses the same grammar.